pkg/actuators/machine/utils.go

```diff
@@ -181,7 +182,12 @@ func providerConfigFromMachine(client client.Client, machine *machinev1.Machine,
 	}
 
 	var config providerconfigv1.AWSMachineProviderConfig
-	if err := codec.DecodeProviderSpec(&machinev1.ProviderSpec{Value: &providerSpecRawExtention}, &config); err != nil {
+
+	//	if err := codec.DecodeProviderSpec(&machinev1.ProviderSpec{Value: &providerSpecRawExtention}, &config); err != nil {
```

Comment:
Can you add to do here explaining why is needed?

Reply:
sure

Reply:
The same here. Please update `DecodeProviderSpec` itself [1] rather than replacing `DecodeProviderSpec`.

[1] https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/apis/awsproviderconfig/v1alpha1/register.go#L79